libs/libc/netdb/lib_dnsaddserver.c

```diff
@@ -45,8 +45,41 @@
 #ifndef CONFIG_NETDB_RESOLVCONF
 /* The DNS server addresses */
 
-union dns_addr_u g_dns_servers[CONFIG_NETDB_DNSSERVER_NAMESERVERS];
-uint8_t g_dns_nservers;    /* Number of currently configured nameservers */
+union dns_addr_u g_dns_servers[CONFIG_NETDB_DNSSERVER_NAMESERVERS] =
+  {
+#if defined(CONFIG_NETDB_DNSSERVER_IPv4)
+    {
+      .ipv4.sin_family      = AF_INET,
```

Comment:
It would be good if we can keep C89 compatible init

Reply:
It's very hard to keep C89 since g_dns_servers's type is union.

Reply:
Yes, I understand that. I do not know what is the best solution. I mean that there is a rule C89 rule for common code, but it is violated quite heavily. I've sked this in the mailing list, but didn't found a solid answer. So C89 compatibility seems to be not a rule, but a semi-rule as it is violated when it is "convenient". I personally do not like semi-rules.

Particularly here reordering of a union members may be applied to get C89 initialization.

I'm fine to keep C89 init if other people find this acceptable

Reply:
> Yes, I understand that. I do not know what is the best solution. I mean that there is a rule C89 rule for common code, but it is violated quite heavily. I've sked this in the mailing list, but didn't found a solid answer. So C89 compatibility seems to be not a rule, but a semi-rule as it is violated when it is "convenient". I personally do not like semi-rules.
> 
> Particularly here reordering of a union members may be applied to get C89 initialization.
> 

Could you give more instruction here? I think the reorder can't fix the ambiguousness for union's field.

> I'm fine to keep C89 init if other people find this acceptable

The initialization is very fragile. Actually, netdb code is in the broken state just because an unrelated change: https://github.com/apache/incubator-nuttx/pull/6770.
We need fix the root problem to avoid it break again in the furture.

Reply:
Yeah... My bad. The union fields rely on `CONFIG_NET_IPvN`, but init code on `CONFIG_NETDB_xxxx`, so reordering will not help here. If those rely on the same config parameters then we could reorder members to get needed members in the first place and use C89 init. Unfortunately that is not the case here

Reply:
So, I would suggest that we accept a little violation here until we find a better initialization method.